components/sst_importer/src/sst_importer.rs

```diff
@@ -698,21 +707,40 @@ impl fmt::Debug for ImportPath {
 pub struct ImportFile {
     meta: SstMeta,
     path: ImportPath,
-    file: Option<File>,
+    file: Option<Box<dyn Write + Send>>,
     digest: crc32fast::Hasher,
+    key_manager: Option<Arc<DataKeyManager>>,
 }
 
 impl ImportFile {
-    fn create(meta: SstMeta, path: ImportPath) -> Result<ImportFile> {
-        let file = OpenOptions::new()
-            .write(true)
-            .create_new(true)
-            .open(&path.temp)?;
+    fn create(
+        meta: SstMeta,
+        path: ImportPath,
+        key_manager: Option<Arc<DataKeyManager>>,
+    ) -> Result<ImportFile> {
+        let file: Box<dyn Write + Send> = if let Some(ref manager) = key_manager {
+            // key manager will truncate existed file, so we should check exist manually.
+            if path.temp.exists() {
+                return Err(Error::Io(io::Error::new(
+                    io::ErrorKind::AlreadyExists,
+                    format!("file already exists, {}", path.temp.to_str().unwrap()),
+                )));
+            }
```

Comment:
Does it break restoring from a checkpoint where there are some incomplete files left in TiKV nodes?

Reply:
The following original implementation with `File` returns an error if the file exists. 
See: https://github.com/tikv/tikv/blob/187774ed9433af80d117b6576179515b2e24ab6a/components/sst_importer/src/sst_importer.rs#L706-L717

Reply:
Lightning never reuses the same SSTMeta name, even when using checkpoints.